src/main/java/org/breedinginsight/brapps/importer/daos/BrAPIObservationUnitDAO.java

```diff
@@ -61,9 +61,23 @@ public BrAPIObservationUnitDAO(BrAPIProvider brAPIProvider) {
         // TODO: Select for study as well
         return observationUnits;
     }
+     */
 
-    public List<BrAPIObservationUnit> createBrAPIObservationUnits(List<BrAPIObservationUnit> brAPIObservationUnitList) throws ApiException {
-        ObservationUnitsApi api = brAPIProvider.getObservationUnitApi(BrAPIClientType.CORE);
-        return BrAPIDAOUtil.post(brAPIObservationUnitList, api::observationunitsPost);
+    public List<BrAPIObservationUnit> getObservationUnitByName(List<String> observationUnitNames, UUID programId) throws ApiException {
+        BrAPIObservationUnitSearchRequest observationUnitSearchRequest = new BrAPIObservationUnitSearchRequest();
+        // could also add programId but
+        observationUnitSearchRequest.observationUnitNames(observationUnitNames);
+        observationUnitSearchRequest.setPageSize(BrAPIDAOUtil.RESULTS_PER_QUERY);
```

Comment:
Ya, what happens if you don't specify a page size on the post search. Is it ignored anyways by the brapi service and pageSize is only relevant for getting the results? If that's the case I think we can remove it. As long as posting without it doesn't default subset the search results that you can get to a certain size.

Reply:
> Is it ignored anyways by the brapi service and pageSize is only relevant for getting the results?

That's how I would implement it, but that doesn't mean that's the right and accepted way :P Maybe we need to check with Pete? We should also check BreedBase.

Reply:
ya, I removed setting the pageSize and at least with the number of objects I'm working with it works with the brapi test server

Reply:
looking at `save_results` in `AJAX/BrAPI.pm`:

```
#set default value to 100000 to get as much as possible records when page size is not a parameter
    if(!$search_params->{pageSize}) {
    	$brapi_module->{page_size} = 100000;
	}
```

So it will default to a large number of records

Reply:
For brapi, the spec doesn't have pageSize as a query parameter for the POST /search endpoints. I guess that might be weird for the immediate response flow but wouldn't matter for the saved search flow that breedbase uses.